agent/lib/kontena/websocket_client.rb

```diff
@@ -126,10 +129,20 @@ def send_request(id, method, params)
     # @param [Faye::WebSocket::API::Event] event
     def on_open(event)
       ping_timer.cancel if ping_timer
-      info 'connection established'
-      @connected = true
+
+      server_version = ws.headers['Kontena-Version']
+
       @connecting = false
-      notify_actors('websocket:open', event)
+      connecting_dt = Time.now - @connecting_at
+      @connected = true
+      @connected_at = Time.parse(@ws.headers['Kontena-Connected-At'])
```

Comment:
This is not backwards-compatible with old servers... they will still accept the request and open up the websocket without sending any `Kontena-Connected-At` headers. Need to deal with that case... likely to end up in a `on :close, code: 4010` very soon.